Product List: make the Products API request ordered by ascending name

jaclync opened the pull request:
Fixes #1569 

## Summary

Since we sort the Products by name on the Products tab, we should also make the API request to fetch a page of Products ordered by ascending name so that the paging is continuous.

## Changes

- Added 2 params to `ProductsRemote`: `orderBy` and `order`
- Updated the API call in `ProductStore`

## Testing

Note: it'd be easier to test for a store with more than 30 Products.

- Log out and in (so that there are no Products on the Products tab to begin with)
- Go to the Products tab and scroll down the Product list --> the Products for each page should be ordered by ascending Product name (e.g. the first page should start from "a"/"A", --> "b"/"B" etc. if any)

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

jaclync committed:
Add `orderBy` and `order` parameters to `ProductsRemote`.

mindgraffiti reviewed (commented):
The updated API request looks good!

Is this a known bug? The results controller either does not alphabetize correctly or the table results are not inserting correctly. The API response show "Spica" in Julia's test store as the first result when I build a request in Postman, and the "test test test" product should display right before Ticket.

Alphabetical ordering in site:
<img width="1018" alt="Screen Shot 2019-12-09 at 10 25 40 AM" src="https://user-images.githubusercontent.com/1062444/70455073-3faab480-1a71-11ea-8fef-9c5e7d2f4d15.png">

Alphabetical ordering on Products screen:
<img src="https://user-images.githubusercontent.com/1062444/70455069-3b7e9700-1a71-11ea-8fef-605cb85257ef.png" width="350" />

Animated GIF was too large to attach here, so I uploaded it to Droplr:
[Animated GIF](https://d.pr/i/vYVavR)

Either way, 🚢 this one!

pmusolino commented:
I noticed that when I update the product title, the product is not inserted correctly in the list, but ends up as the last in the list, not following the ascending order.
![Simulator Screen Shot - iPhone 11 Pro - 2019-12-09 at 18 00 18](https://user-images.githubusercontent.com/495617/70456164-e4e37e00-1aad-11ea-8141-8a6dea5e02ee.png)

jaclync committed:
Sort Product names with `localizedCompare`.

jaclync commented:
@pmusolino @mindgraffiti great catch! I updated some Product names on my test store to reproduce this issue. It feels like the default String comparison is based on the string's ASCII values (so that lower case goes after upper case letters). I checked the [Apple documentation about swift String](https://developer.apple.com/documentation/swift/string), and the doc says:

> Comparing strings using the equal-to operator (==) or a relational operator (like < and >=) is always performed using the Unicode canonical representation, so that different representations of a string compare as being equal.

and on [NSString's `compare` doc](https://developer.apple.com/documentation/foundation/nsstring/1414082-compare), there is a discussion about comparing two `NSString`s:

> Important
When working with text that’s presented to the user, use the localizedStandardCompare(_:) method instead.

Therefore, I updated the `NSSortDescriptor` to compare the Product names with `localizedStandardCompare(_:)`: https://github.com/woocommerce/woocommerce-ios/pull/1586/commits/7a835253d19b110a3be69b51b5b623aa68818cb2

Even though the order is still slightly inconsistent with wp-admin when sorting the Products by name, the "Spica" in Julia's test store and both lower and upper case letters should be sorted together. It also takes the device locale into consideration.

app | wp-admin
-- | --
![Simulator Screen Shot - iPhone 8 - 2019-12-10 at 21 51 23](https://user-images.githubusercontent.com/1945542/70535116-6ebb4600-1b97-11ea-8674-b7d38beee540.png) | <img width="287" alt="Screen Shot 2019-12-10 at 9 51 42 PM" src="https://user-images.githubusercontent.com/1945542/70535115-6e22af80-1b97-11ea-95fa-232419a40e26.png">

Please check again if the order makes more sense now!

mindgraffiti reviewed (approved):
Nice! I stand by my original 🚢 it

pmusolino reviewed (approved):
Now it works like a charm :shipit:
